api/src/services/setup/utils.js

```diff
@@ -291,6 +294,42 @@ const unstageStagedDdocs = async () => {
   }
 };
 
+const getUpgradeServicePayload = (stagingDoc) => {
+  const dockerCompose = {};
+  const attachmentPrefix = 'docker-compose/';
+
+  Object
+    .entries(stagingDoc._attachments)
+    .filter(([name]) => name.startsWith(attachmentPrefix))
+    .forEach(([name, contents]) => {
+      const buffer = Buffer.from(contents.data, 'base64');
+      const fileName = name.replace(attachmentPrefix, '');
+      dockerCompose[fileName] = buffer.toString('utf-8');
+    });
+
+  const tags = stagingDoc.tags.map(tag => ({
+    containerName: tag.container_name,
+    imageTag: tag.image,
+  }));
+
+  return {
+    tags,
+    'docker-compose': dockerCompose,
+  };
+};
+
+const makeUpgradeRequest = (payload) => {
+  let url;
+  try {
+    url = new URL(UPGRADE_SERVICE_URL);
+    url.pathname = '/upgrade';
+  } catch (err) {
+    throw new Error(`Invalid UPGRADE_SERVICE_URL: ${UPGRADE_SERVICE_URL}`);
+  }
+
+  return rpn.post({ url: `${UPGRADE_SERVICE_URL}/upgrade`, json: true, body: payload });
```

Comment:
Surely you can use the URL you created above?

```suggestion
  return rpn.post({ url, json: true, body: payload });
```

Reply:
Oh, that was the intention .. and then life happened :)) 
Thanks for noticing!